agent/engine/task_manager.go

```diff
@@ -1195,11 +1208,6 @@ func (mtask *managedTask) discardEvents() {
 		case <-mtask.acsMessages:
 		case <-mtask.resourceStateChangeEvent:
 		case <-mtask.ctx.Done():
```

Comment:
It's OK to never close a channel in Golang. The only reason to close a channel is when a receiver is looking for a close signal. In this case, we only send messages to these channels so closing them can only result in a potential panic.